modules/charon-core/src/main/java/org/wso2/charon3/core/attributes/ComplexAttribute.java

```diff
@@ -71,10 +73,22 @@ public Attribute getSubAttribute(String attributeName) throws CharonException {
      * @throws CharonException
      */
     @Override
-    public void deleteSubAttributes() throws CharonException {
+    public void deleteSubAttributes() {
```

Comment:
We are not recommending to change public methods as we need to preserve the backward compatibility as these public methods can be used from different other places.